topostats/tracing/splining.py

```diff
@@ -587,13 +591,14 @@ def splining_image(
                 }
                 molstats[grain_no.split("_")[-1] + "_" + mol_no.split("_")[-1]] = {
                     "image": filename,
-                    "grain_number": grain_no.split("_")[-1],
+                    "grain_number": int(grain_no.rsplit("_", maxsplit=1)[-1]),
```

Comment:
`maxsplit=1` looks redundant if you're only ever interested in the last element of the returned list as seems to be the case here and on the next line. There could be five `_` but the `[-1]` always picks up the last value.

```python
In [15]: "1_2_3_4_5_6".rsplit("_")[-1]
Out[15]: '6'

In [16]: "1_2_3_4_5_6".rsplit("_", maxsplit=1)[-1]
Out[16]: '6'

In [17]: "1_2_3_4_5_6".split("_")[-1]
Out[17]: '6'
```
```suggestion
                    "grain_number": int(grain_no.split("_")[-1]),
```

To which end you could also use `split` as is the case on line 597 as `rsplit()` is only really required when you do need to use `maxsplit`.

Same for next line too.

Reply:
Is it [C0207](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/use-maxsplit-arg.html)?

If so the comment suggests we may be able to disable it as there aren't going to be a huge number of splits in this case (just one to separate `grain_mol` I think).

> Be aware that the performance improvement from not splitting the string so many times will only be realized in cases presenting more instances of the splitting character than the minimal example here.

Reply:
Yep and cool, I've reverted and added the pylint: disable comment so it passes